kvs/src/kvs/rep_factor_response_handler.cpp

```diff
@@ -135,24 +137,33 @@ void rep_factor_response_handler(
             auto res = process_get(key, serializer);
             tp->set_value(res.first.reveal().value);
             tp->set_error(res.second);
-
-            key_access_timestamp[key].insert(std::chrono::system_clock::now());
-            total_access += 1;
           } else {
             auto time_diff =
                 std::chrono::duration_cast<std::chrono::milliseconds>(
                     now - start_time)
                     .count();
             auto ts = generate_timestamp(time_diff, wt.get_tid());
 
-            process_put(key, ts, request.value_, serializer, key_size_map);
+            process_put(key, ts, request.value_, serializer);
             tp->set_error(0);
 
-            key_access_timestamp[key].insert(now);
-            total_access += 1;
-            local_changeset.insert(key);
+            if (kSelfTierId != 3 || (kSelfTierId == 3 && wt.get_tid() == 0)) {
```

Comment:
Is there a way to avoid having to repeat these checks? Maybe by moving to the front of the function?

Reply:
it's kinda tricky...
We do this check whenever we have PUT, but there is an outer branch condition on `responsible && request.addr_ `